frontend/src/app/util/snap_data.cljs

```diff
@@ -0,0 +1,245 @@
+;; This Source Code Form is subject to the terms of the Mozilla Public
+;; License, v. 2.0. If a copy of the MPL was not distributed with this
+;; file, You can obtain one at http://mozilla.org/MPL/2.0/.
+;;
+;; Copyright (c) UXBOX Labs SL
+
+(ns app.util.snap-data
+  "Data structure that holds and retrieves the data to make the snaps. Internaly
+   is implemented with a balanced binary tree that queries by range.
+   https://en.wikipedia.org/wiki/Range_tree"
+  (:require
+   [app.common.data :as d]
+   [app.common.pages :as cp]
+   [app.common.pages.diff :as diff]
+   [app.common.uuid :as uuid]
+   [app.util.geom.grid :as gg]
+   [app.util.geom.snap-points :as snap]
+   [app.util.range-tree :as rt]))
+
+(def snap-attrs [:frame-id :x :y :width :height :hidden :selrect :grids])
+
+;; PRIVATE FUNCTIONS
+
+(defn- make-insert-tree-data
+  [shape-data axis]
+  (fn [tree]
+    (let [tree (or tree (rt/make-tree))]
+      (as-> tree $
```

Comment:
for a single operation I think `as->` is useless here and in the following functions

Reply:
OK